src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -2806,5 +2806,50 @@ static void Main()
             Assert.Single(diagnostic.AdditionalLocations);
         }
 
+        [Fact]
+        public void CopyConstructor_01()
+        {
+            var source = """
+                #nullable enable
+                var r = new Rec(); // to show that implicit parameterless ctor still exists, hence diagnostic 1
+
+                record Rec
+                {
+                    public string Prop { get; init; } // 1
```

Comment:
Do we need a test if the property is marked as `required`?

Reply:
Yeah, we'll want to test. We might be treating copy constructors as implicitly having SetsRequiredMembers.